buffer/src/main/java/io/netty/buffer/ByteBufProcessor.java

```diff
@@ -16,10 +16,14 @@
 
 package io.netty.buffer;
 
-public interface ByteBufProcessor {
+/**
+ * Use {@link ByteVisitor} directly where interfaces permit.
+ */
+public interface ByteBufProcessor extends ByteVisitor {
```

Comment:
> Moving ByteString, AsciiString and ByteVisitor to netty-common (since it's not limited to writing a codec anymore)

@trustin - We may be creating a circular dependency here.  `ByteString` exposes methods that take `ByteBuf` types...and so we would have to add `netty-buffer` as a dependency of `netty-common` but the `netty-buffer` package already depends upon `netty-common`.  We may have to introduce some an abstraction layer...or move some other stuff....any suggestions?